### PR TITLE
Reconnect on failures rabbitmq

### PR DESCRIPTION
**Pull request by ankurjain86**

### CODE REVIEW

File: examples/app/config/test/service.yml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 name: 'service'
 messages:
   host: 'amqp://localhost'
+  retryDelay: 100
```
Comment (igorshapiro) [inline examples/app/config/test/service.yml]: Let's make it 5 seconds?

File: examples/app/test/pubsub/pubsub_consumer_test.js [hunk 2]
```diff
@@ -104,5 +104,41 @@ describe('pubsub', function() {
       expect(deadLetterEntries[0]).to.containSubset({attemptsMade: 6, maxAttempts: 5, messageType: 'orders.test.consumer', content: {hello: 'world'}})
       expect(deadLetterEntries[0].killedAt).to.not.be.null
     })
+
+    it.only('test reconnect', async function() {
```
Comment (igorshapiro) [inline examples/app/test/pubsub/pubsub_consumer_test.js]: Remove `only`

File: examples/app/config/test/service.yml (hunk 1 above)
Comment (igorshapiro) [reply examples/app/config/test/service.yml]: Ah, got it. If you do it for test, it's better to mock `service.messages.retryDelay` and use sinon fake timers: http://sinonjs.org/releases/v3.2.1/fake-timers/

File: examples/app/test/pubsub/pubsub_consumer_test.js [hunk 3]
```diff
@@ -104,5 +104,41 @@ describe('pubsub', function() {
...
+      await channel.assertExchange('orders.test.consumer', 'fanout', {durable: false})
+      await channel.bindQueue(QUEUE_NAME, 'orders.test.consumer', '')
+      let handler = {
+        handle: this.sinon.spy()
+      }
+      hubMessageProcessor.register('orders.test.consumer', handler)
+
+      // Send a message through pub sub
+      pubsub.publish('orders.test.consumer', { hello: 'world' })
+      await Promise.delay(200)
```
Comment (igorshapiro) [inline examples/app/test/pubsub/pubsub_consumer_test.js]: Here it's better to use fake timers instead of actual delay

Comment (ankurjain86) [reply examples/app/test/pubsub/pubsub_consumer_test.js]: But this wait is to send message to the queue and wait for it to be processed by consumer. Not sure if fake timers would help there.

File: services/pubsub/rabbit_message_adapter.js [hunk 4]
```diff
@@ -5,9 +5,12 @@ module.exports = class RabbitMqMessageAdapter {
     this.log = log
     this.config = config
     this.messageProcessor = messageProcessor
+    this.retryDelay = this.config.get('service.messages.retryDelay', 5000)
```
Comment (ankurjain86) [inline services/pubsub/rabbit_message_adapter.js]: Set default retry to 5 sec

File: examples/app/test/pubsub/pubsub_consumer_test.js (hunk 3 above)
Comment (igorshapiro) [reply examples/app/test/pubsub/pubsub_consumer_test.js]: Sorry, my bad. Let's just add the 

```js
QT.stubConfig('service.messages.retryDelay', 100)
```

Instead putting it in the test config (as it the configuration implicit in this test)

Comment (ankurjain86) [reply examples/app/test/pubsub/pubsub_consumer_test.js]: I tried putting a stub config. If i use stub then test cases fails, else everything passes. Also it fails when i remove config from service.yml as it defaults to 5seconds. I need to debug further on whats going on

Comment (igorshapiro) [reply examples/app/test/pubsub/pubsub_consumer_test.js]: Then let's merge it and publish now, and later we'll figure out what's the problem with stubConfig

**Pull request merged**